scheduler/reconcile_util.go

```diff
@@ -225,8 +225,20 @@ func (a allocSet) filterByTainted(taintedNodes map[string]*structs.Node, support
 	ignore = make(map[string]*structs.Allocation)
 
 	for _, alloc := range a {
-		// Terminal allocs are always untainted as they should never be migrated.
-		if alloc.TerminalStatus() {
+		reconnected := false
+		expired := false
+
+		// Only compute reconnected for unknown, running, and failed since they need to go through the reconnect logic.
+		if supportsDisconnectedClients &&
+			alloc.ClientStatus != structs.AllocClientStatusPending &&
+			alloc.ClientStatus != structs.AllocClientStatusComplete &&
+			alloc.ClientStatus != structs.AllocClientStatusLost {
+			reconnected, expired = alloc.Reconnected()
+		}
```

Comment:
I don't love the chain of "not-and" rather than "or", but not a blocker.

Reply:
Gotcha. I misunderstood and thought you were talking about the comment in my haste. I'll fix that up tomorrow.